odl/test/tomo/operators/ray_trafo_test.py

```diff
@@ -237,21 +236,61 @@ def test_adjoint(projector):
     assert result_AxAx == pytest.approx(result_xAtAx, rel=rtol)
 
 
+def test_adjoint_of_adjoint(projector):
+    """Test Ray transform adjoint of adjoint."""
+
+    # Create Shepp-Logan phantom
+    vol = odl.phantom.shepp_logan(projector.domain, modified=True)
+
+    # Calculate projection
+    proj = projector(vol)
+    proj_adj_adj = projector.adjoint.adjoint(vol)
+
+    # Verify A(x) == (A^*)^*(x)
+    assert all_almost_equal(proj, proj_adj_adj)
+
+    # Calculate adjoints
+    proj_adj = projector.adjoint(proj)
+    proj_adj_adj_adj = projector.adjoint.adjoint.adjoint(proj)
+
+    # Verify A^*(y) == ((A^*)^*)^*(x)
+    assert all_almost_equal(proj_adj, proj_adj_adj_adj)
+
+
 def test_angles(projector):
-    """Test discrete Ray transform angle conventions."""
-    # Smoothed line/hyperplane around 0:th dimension
-    vol = projector.domain.element(lambda x: np.exp(-x[0] ** 2))
+    """Test Ray transform angle conventions."""
+
+    # Smoothed line/hyperplane with offset
+    vol = projector.domain.element(
+        lambda x: np.exp(-(2 * x[0] - 10 + x[1]) ** 2))
 
     # Create projection
     result = projector(vol).asarray()
 
-    # Find the angle where the projection has a maximum (along the line)
+    # Find the angle where the projection has a maximum (along the line).
     axes = 1 if projector.domain.ndim == 2 else (1, 2)
     ind_angle = np.argmax(np.max(result, axis=axes))
-    maximum_angle = projector.geometry.angles[ind_angle]
-
-    # We expect the maximum at pi / 2 (and possibly multiples of it)
-    assert almost_equal(np.fmod(maximum_angle, np.pi), np.pi / 2, places=1)
+    # Restrict to [0, 2 * pi) for helical
+    maximum_angle = np.fmod(projector.geometry.angles[ind_angle], 2 * np.pi)
+
+    # Verify correct maximum angle. The line is defined by the equation
+    # x1 = 10 - 2 * x0, i.e. the slope -2. Thus the angle arctan(1/2) should
+    # give the maximum projection values.
+    expected = np.arctan2(1, 2)
+    assert np.fmod(maximum_angle, np.pi) == pytest.approx(expected, abs=0.1)
+
+    # Find the pixel where the projection has a maximum at that angle
```

Comment:
I guess changing these to center of mass would wastly improve the accuracy of these things. At least make a TODO here that we should do that for later.